docs/tutorials/php-guide.md

```diff
@@ -0,0 +1,322 @@
+should update!
+
+![example file](images/php-rename-post.png "example file")
+
+#### Code Actions
+
+Code action support isn't great, in either Intelephense or phpactor. But there
+exists a few code actions, for example it's possible to quickly provide a
+docblock for a function.
+
+With the cursor on the `test_fn` declaration, invoke `lsp-execute-code-action`
+(`s-l a a`). There should now be a simple docblock for the function.
+
+![example file](images/php-code-action-post.png "example file")
+
+### Navigation
+`lsp-mode` has integration with `xref` core package and in addition it has
+
+- `xref-find-definitions`(`M-.` and `s-l g g`) - find definition(s) at point
+- `xref-find-references`(`s-l g r`) - find references to the symbol at point
+- `lsp-treemacs-type-hierarchy` - show type hierarchy
+
+Depending on which selection framework(s) you use, there should also be packages
+available to support it.
+
+For example:
+
+- [helm-lsp](https://github.com/emacs-lsp/helm-lsp)
+- [lsp-ivy](https://github.com/emacs-lsp/lsp-ivy)
+- [consult-lsp](https://github.com/gagbo/consult-lsp)
+
+### [lsp-treemacs](https://github.com/emacs-lsp/lsp-treemacs)
+`lsp-treemacs` is a package providing integration with
+[treemacs](https://github.com/Alexander-Miller/treemacs) and an alternative
+tree-view visualization. Refer to the project's readme for further information.
+
+![tree references](images/php-treemacs-references.png "Treemacs references")
+
+### Help/Documentation
+`lsp-mode` automatically enables `eldoc-mode` which will show hover information
+in the minibuffer. `lsp-mode` is showing only the one line signature info so if
+you want to see the full documentation at point you can use
+`lsp-describe-thing-at-point`.
+
+![hover](images/php-hover-info.png)
+
+In addition to that you can also see signature help after pressing `C-M-SPC` or
+after pressing trigger char like `(`. If there is more than one applicable
+signature due to function overloading, you may browse between the available
+candidates using `M-n/M-p`.  may browse them via `M-n/M-p`
+
+![signature info](images/php-signature-help.png)
+
+### Diagnostics
+For on-the-fly errors `lsp-mode` is using `flycheck` (`flymake` is also
+supported). It is configured automatically. In addition to standard `flycheck`
+features `lsp-mode` provides a project-wide error list via
+`lsp-treemacs-errors-list`.
+
+Depending on which selection framework(s) you use, there should also be packages
+available to support it.
+
+For example:
+
+- [helm-lsp](https://github.com/emacs-lsp/helm-lsp)
+- [lsp-ivy](https://github.com/emacs-lsp/lsp-ivy)
+- [consult-lsp](https://github.com/gagbo/consult-lsp)
+
+![errors list](images/php-errors-list.png)
+
+## Debugging
+
+Much like `lsp` is to code navigation/refactoring, there exists a similar spec
+called `dap` ([debug adapter
+protocol](https://microsoft.github.io/debug-adapter-protocol/)) for debugging.
```

Comment:
Wonder whether this will be rendered correctly with link text split into 2 lines)

Reply:
It should do, but I think it's just `gq` weirdness. I'll correct it.